src/logger.c

```diff
@@ -40,59 +40,62 @@ static const char *logident = NULL;
 bool logcontrol = false;
 int umbilical = 0;
 
+static bool should_log(int level) {
+	// Bail out early if there is nothing to do.
+
+	if(!logcontrol && (level > debug_level || logmode == LOGMODE_NULL)) {
+		return false;
+	}
+
+	return level <= debug_level;
```

Comment:
This turns out to be wrong too. If `logcontrol == true`, the value of `level` is not necessarily the same debug level requested over a control channel. So we should `return true` here, or just simplify this whole function to:

```
return (level <= debug_level && logmode != LOGMODE_NULL) || logcontrol;
```

Reply:
Updated with the code you provided.